src/vivarium_gates_nutrition_optimization_child/components/causes.py

```diff
@@ -1,17 +1,81 @@
 """
 Component to include birth prevalence in SIS model.
 """
-
+import pandas as pd
 from vivarium.framework.state_machine import State
-from vivarium_public_health.disease import DiseaseModel, DiseaseState
+from vivarium_public_health.disease import DiseaseModel as DiseaseModel_
+from vivarium_public_health.disease import DiseaseState
 from vivarium_public_health.disease import (
     RiskAttributableDisease as RiskAttributableDisease_,
 )
 from vivarium_public_health.disease import SusceptibleState
 from vivarium_public_health.disease.transition import TransitionString
 
 
-def SIS_with_birth_prevalence(cause: str) -> DiseaseModel:
+class DiseaseModel(DiseaseModel_):
+    @property
+    def columns_required(self):
+        return ["age", "sex", "alive", "tracked"]
+
+    def setup(self, builder):
+        super().setup(builder)
+        if "variable_step_sizes" in self._get_data_functions:
+            for state, step_size in self._get_data_functions["variable_step_sizes"].items():
+                builder.time.register_step_modifier(
+                    lambda index: self.modify_step(state, step_size, index)
+                )
+
+    def modify_step(self, state, step_size, index):
+        infected = self.population_view.get(
+            index,
+            f"{self.state_column} == '{state}' and alive == 'alive' and tracked == True",
+        ).index
+        return pd.Series(pd.Timedelta(days=step_size), index=infected)
+
+
+def SIS(cause: str, step_size_days: str = None) -> DiseaseModel:
```

Comment:
In the interest of wrapping up this model, this is fine for now, but a better approach would be to be specifying step-size values as configuration values and having that code inside `DiseaseModel` itself rather than as an argument to the various implementations. That would reduce code duplication, make the configuration easier to understand, and most significantly enable a user to set a step-size for any state in any disease model. 

That being said, if we are moving towards a method of determining step-sizes that is more sophisticated than having the user define them all individually, this approach will need to be revisited.